py/client/pydeephaven/ticket.py

```diff
@@ -0,0 +1,139 @@
+    closed. Many types of server objects are exportable and can be fetched to the client as an export ticket or as an
+    instance of wrapper class (such as :class:`table.Table`, :class:`plugin_client.PluginClient`，etc. that wraps the
+    export ticket. An export ticket can be published to a :class:`.SharedTicket` so that the exported server object can
+    be shared with other sessions.
+
+    Note: Users should not create ExportTicket objects directly. They are managed by the Session object and are
+    automatically created when exporting objects from the server via. :meth:`.Session.open_table`, :meth:`.Session.fetch`,
+    and any operations that returns a Table object.
+    """
+
+    def __init__(self, ticket_bytes: bytes):
+        """Initializes an ExportTicket object
+
+        Args:
+            ticket_bytes (bytes): the raw bytes for the ticket
+        """
+        if not ticket_bytes:
+            raise DHError('ExportTicket: ticket is None')
+        elif not ticket_bytes.startswith(b'e'):
+            raise DHError(f'ExportTicket: ticket {ticket_bytes} is not an export ticket')
+
+        super().__init__(ticket_bytes)
+
+
+class ScopeTicket(Ticket):
+    """A ScopeTicket is a ticket that references a scope variable on the server. Scope variables are ones in the global
+    scope of the server and are accessible to all sessions. Scope variables can be fetched to the client as an export
+    ticket or a Deephaven :class:`table.Table` that wraps the export ticket. """
+
+    def __init__(self, ticket_bytes: bytes):
+        """Initializes a ScopeTicket object
+
+        Args:
+            ticket_bytes (bytes): the raw bytes for the ticket
+        """
+        if not ticket_bytes:
+            raise DHError('ScopeTicket: ticket is None')
+        elif not ticket_bytes.startswith(b's/'):
+            raise DHError(f'ScopeTicket: ticket {ticket_bytes} is not a scope ticket')
+
+        super().__init__(ticket_bytes)
+
+    @classmethod
+    def scope_ticket(cls, name: str) -> ScopeTicket:
+        """Creates a scope ticket that references a scope variable by name.
+
+        Args:
+            name (str): the name of the scope variable
+
+        Returns:
+            a ScopeTicket object
+        """
+        return cls(ticket_bytes=f's/{name}'.encode(encoding='ascii'))
+
+
+def _ticket_from_proto(ticket: ticket_pb2.Ticket) -> Ticket:
+    """Creates a Ticket object from a gRPC protobuf ticket object.
+
+    Args:
+        ticket (ticket_pb2.Ticket): the gRPC protobuf ticket object
+
+    Returns:
+        a Ticket object
+
+    Raises:
+        DHError: if the ticket type is unknown
+    """
+    ticket_bytes = ticket.ticket
+    if ticket_bytes.startswith(b'h'):
+        return SharedTicket(ticket_bytes)
+    elif ticket_bytes.startswith(b'e'):
+        return ExportTicket(ticket_bytes)
+    elif ticket_bytes.startswith(b's/'):
+        return ScopeTicket(ticket_bytes)
+    else:
```

Comment:
I do expect that new types of tickets to be created by hand because I assume that when we have the need to support a new category of objects (represented by a new ticket type), we'll need to extend the API somewhat even if the client is only required to support publishing/fetching the new type of tickets.